### PR TITLE
#54 Add column info variant

### LINKED ISSUE #54: FR: Columns Info variation
Variation block - To support different content styling in each column, icons with alignment options and 'red line' on top. CTAs to align with generic ones

Design: 

https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2243-164763&mode=dev

https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?type=design&node-id=2220-83238&mode=dev

Image

Image

Image

### Requirements
- Create a variation of the columns block that has the styling in the [design] (Variation name: Columns (info) )(https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?node-id=2220%3A83239&mode=dev), as follows:
  - possibility to add a column with an H2 and a text, that would show the column in the style of the first column (with a red line on top)
  - Possibility to add additional columns in the style of the second / third column (icon, H5, text, button regular)
  - The columns' order can be switched, to have the heading column in the middle, for example, or no heading column at all, depending on the author's input

### Open questions:
- How many H2 can be added? - responsibility of the authors, in the designs there is never more than 1 atm
- Image as a bg - out of scope for this ticket, handled in a separate one

### PR DESCRIPTION
**Pull request by TomaszDziezykNetcentric**
Fix #54 

Test URLs:
- Before: https://main--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/tdziezyk/v2/columns
- After: https://54-columns-info--vg-macktrucks-com-rd--netcentric.hlx.page/drafts/tdziezyk/v2/columns

### CODE REVIEW

**Review by Lakshmishri — DISMISSED** [review]
- The gap between icon and text should be 16px as per [designs](https://www.figma.com/file/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?node-id=2220%3A83250&mode=dev)

- The links/buttons should be aligned
- At 533px they don't look aligned
Screenshot 2023-09-08 at 15 00 48

File: blocks/v2-columns/v2-columns.js [hunk 1]
```diff
@@ -42,14 +45,46 @@ export default async function decorate(block) {
...
     }
   });
+
+  // logic for info variant
+  if (block.classList.contains(`${blockName}--info`)) {
+    const headings = [...block.querySelectorAll('h3, h4, h5, h6')];
+    const h2List = [...block.querySelectorAll('h2')];
+
+    headings.forEach((h) => {
+      h.classList.add('h5');
```
Comment (kesiah) [inline blocks/v2-columns/v2-columns.js:74]: Careful, for some reason you are adding 2 classes in the same item
image

File: blocks/v2-columns/v2-columns.css [hunk 2]
```diff
@@ -20,6 +20,12 @@
   display: block;
 }
 
+.v2-columns__column--with-text .icon,
```
Comment (kesiah) [inline blocks/v2-columns/v2-columns.css:23]: Btw mobile and tablet, it looks really strange:
image

Comment (kesiah) [reply blocks/v2-columns/v2-columns.css:23]: In tablet view, I would say, the most important element is the smallest, is that expected @cogniSyb ?
image

**Review by kesiah — CHANGES_REQUESTED** [review]
Could you include an example, where 3 items are the same? I mean, 3 columns with icons.

File: blocks/v2-columns/v2-columns.css (hunk 2 above)
Comment (TomaszDziezykNetcentric) [reply blocks/v2-columns/v2-columns.css:23]: Now they have the same width

**Review by kesiah — APPROVED** [review]

**Review by markovukiceviccn — APPROVED** [review]

**Pull request merged**